staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -495,6 +497,12 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 	if a.group.MetaGroupVersion != nil {
 		reqScope.MetaGroupVersion = *a.group.MetaGroupVersion
 	}
+	if a.group.OpenAPIConfig != nil {
+		openAPIDefinitions, err := openapibuilder.BuildOpenAPIDefinitionsForResource(defaultVersionedObject, a.group.OpenAPIConfig)
+		if err == nil {
```

Comment:
And if this error is not nil, then what? We should fail everything with a clear error?

Reply:
That makes sense to me too, When I do that it fails to install the CRD handler for the extension apiserver, I think that the extension apiserver doesn't serve the openapi spec for CRDs themselves, but I'm looking into it to make sure.

Also, a user defined CR might not have an openapi spec, and in that case do we still want the handler install to fail?

Reply:
the issue is that the CRD openapi spec doesn't have the "x-kubernetes-group-version-kind" property for some reason

Reply:
I suspect that kubectl apply on a CRD doesn't work as it is now, since it relies on "x-kubernetes-group-version-kind" being present, unless some special case logic was implemented in kubectl

Reply:
> When I do that it fails to install the CRD handler for the extension apiserver, I think that the extension apiserver doesn't serve the openapi spec for CRDs themselves, but I'm looking into it to make sure. (Either way this doesn't work for them)

If there's an error there, let's fix it at its source and correct the openapi bit, not ignore the error here.

> a user defined CR might not have an openapi spec, and in that case do we still want the handler install to fail?

I don't think custom resource handlers use this installer. If they do use this installer, and don't have openapi available, I would expect their group meta to not have an OpenAPIConfig set. either way, we shouldn't ignore errors here.

Reply:
> I suspect that kubectl apply on a CRD doesn't work as it is now, since it relies on "x-kubernetes-group-version-kind" being present, unless some special case logic was implemented in kubectl

When `kubectl apply` can't find a schema or go struct, it falls back to generic merge patch computation:

https://github.com/kubernetes/kubernetes/blob/10b8665a1c9dd5714b13130fcbf5c04f7f658178/pkg/kubectl/cmd/apply.go#L717-L728

Reply:
Yes, let's fix the extensions-apiserver.

I'm OK with doing that in a separate PR (and depending on how large of a rabbit hole, I might actually prefer it) if
* that error specifically is caught here
* other errors still cause a failure
* there is a big, sternly worded TODO explaining that the extensions-apiserver is Doing It Wrong
* ...you intend to fix extensions apiserver as like the next thing you do.

Reply:
https://github.com/kubernetes/kubernetes/issues/64169